tests/github/interactive_api_director/experiments/pytorch_kvasir_unet/envoy/kvasir_shard_descriptor.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Kvasir shard descriptor."""
+
+
+import os
+from pathlib import Path
+
+import numpy as np
+from PIL import Image
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+from openfl.utilities import validate_file_hash
+
+
+class KvasirShardDescriptor(ShardDescriptor):
+    """Shard descriptor class."""
+
+    def __init__(self, data_folder: str = 'kvasir_data',
+                 rank_worldsize: str = '1,1',
+                 enforce_image_hw: str = None) -> None:
+        """Initialize KvasirShardDescriptor."""
+        super().__init__()
+
+        self.data_folder = Path.cwd() / data_folder
+        self.download_data(self.data_folder)
+
+        # Settings for resizing data
+        self.enforce_image_hw = None
+        if enforce_image_hw is not None:
+            self.enforce_image_hw = tuple(int(size) for size in enforce_image_hw.split(','))
+        # Settings for sharding the dataset
+        self.rank_worldsize = tuple(int(num) for num in rank_worldsize.split(','))
+
+        self.images_path = self.data_folder / 'segmented-images' / 'images'
+        self.masks_path = self.data_folder / 'segmented-images' / 'masks'
+
+        self.images_names = [
+            img_name
+            for img_name in sorted(os.listdir(self.images_path))
+            if len(img_name) > 3 and img_name[-3:] == 'jpg'
+        ]
+        # Sharding
+        self.images_names = self.images_names[self.rank_worldsize[0] - 1::self.rank_worldsize[1]]
+
+        # Calculating data and target shapes
+        sample, target = self[0]
+        self._sample_shape = [str(dim) for dim in sample.shape]
+        self._target_shape = [str(dim) for dim in target.shape]
```

Comment:
Because `ShardDirectorClient.report_shard_info` expects strings for `sample_shape` and `target_shape`. We use string type here to allow users to specify shape ranges, e. g. if images have different heights or widths we would use `['100-200','300-400']`.